packages/uni_app/lib/view/restaurant/widgets/favorite_restaurants_button.dart

```diff
@@ -0,0 +1,44 @@
+import 'package:flutter/material.dart';
+import 'package:uni/generated/l10n.dart';
+import 'package:uni_ui/icons.dart';
+
+class FavoriteRestaurantsButton extends StatelessWidget {
+  const FavoriteRestaurantsButton({
+    super.key,
+    required this.isFavoriteOn,
+    required this.onToggle,
+  });
+
+  final bool isFavoriteOn;
+  final VoidCallback onToggle;
+
+  @override
+  Widget build(BuildContext context) {
+    return TextButton(
+      onPressed: onToggle,
+      style: TextButton.styleFrom(
+        padding: const EdgeInsets.symmetric(horizontal: 20, vertical: 10),
+        backgroundColor: Theme.of(context).colorScheme.surfaceContainer,
+        shape: RoundedRectangleBorder(
+          borderRadius: BorderRadius.circular(50),
+        ),
+      ),
+      child: Row(
+        mainAxisAlignment: MainAxisAlignment.center,
+        children: [
+          Text(
+            S.of(context).favourite_filter,
+            style: Theme.of(context).textTheme.bodyMedium,
+          ),
+          const SizedBox(width: 8),
+          if (!isFavoriteOn)
+            const UniIcon(
+              UniIcons.heartOutline,
+            )
+          else
+            const UniIcon(UniIcons.heartFill),
```

Comment:
```suggestion
          const UniIcon(
            !isFavoriteOn ? UniIcons.heartOutline : UniIcons.heartFill
          )
```
Prefer verifying the condition using a ternary operator in the argument.